astropy/units/core.py

```diff
@@ -2321,7 +2320,7 @@ def __init__(
                     for p in unit.powers
                 ]
 
-            self._scale = sanitize_scale_value(scale)
+            self._scale = sanitize_scale(scale)
```

Comment:
I thought in principle we should now be able to remove this second sanitization, but then realized that of course after the above multiplications, it is possible a scale becomes very close to unity. But in that case, do we need to do the full sanitization on top? Overall, it is probably fine since the one on top is skipped in internal use.